plugins/BeamerPresentation/main.lua

```diff
@@ -0,0 +1,46 @@
+local m = {
+    dec = 1,
+    inc = 2,
+}
+-- Register all Toolbar actions and intialize all UI stuff
+function initUi()
+    app.registerUi({menu = "beamer +1", callback = "Beamer", mode=m.inc, accelerator="<Alt>y"});
+    app.registerUi({menu = "beamer -1", callback = "Beamer", mode=m.dec, accelerator="<Alt>x"});
+    app.registerUi({menu = "beamer clean", callback = "BeamerClean"});
+end
+
+function Beamer(mode)
+    local struct = app.getDocumentStructure()
+    local diff
+    if mode == m.inc then
+        diff = 1
+    elseif mode == m.dec then
+        diff = -1
+    end
```

Comment:
I would prefer to write this in one line, but it's a matter of style preference.
```suggestion
    local diff = (mode == m.inc) and 1 or -1
```

Reply:
True, the idea was to keep this portion of the code expandable in case a new mode is added (and with three or more cases the oneliner gets pretty hard to read in my opinion). Still, in the current state, the oneliner is nicer I agree (and we can always expand this if adding new modes).